pkg/proto/nbcontract/v1/identityconfig.proto

```diff
@@ -2,16 +2,8 @@ syntax = "proto3";
 package nbcontract.v1;
 
 message IdentityConfig {
-  IdentityType identity_type = 1;
-  optional string service_principal_id = 2;
-  optional string service_principal_secret = 3;
-  optional string assigned_identity_id = 4;  //could be user or system assigned, depending on the type
-  optional string use_managed_identity_extension = 5;
-}
-
-enum IdentityType {
-  IDENTITY_TYPE_UNSPECIFIED = 0;
-  IDENTITY_TYPE_SERVICE_PRINCIPAL = 1;
-  IDENTITY_TYPE_USER_IDENTITY = 2;
-  // IDENTITY_TYPE_SYSTEM_IDENTITY = 3; not supported yet
+  string service_principal_id = 1;
+  string service_principal_secret = 2;
+  string assigned_identity_id = 3;  //could be user or system assigned, depending on the type
+  string use_managed_identity_extension = 4;
```

Comment:
Wondering if this is only used for generating azure config file? If yes - it might make sense to collect all the settings that affect it in one place ...

Reply:
Grouped all the auth config fields and cluster config fields used to generate azure config file as listed here: https://cloud-provider-azure.sigs.k8s.io/install/configs/ into AuthConfig and ClusterConfig.